### PR TITLE
MPS-3051: Team Permission Endpoint and Entities

### PR DESCRIPTION
**Pull request by elhicks**
https://vimean.atlassian.net/browse/MPS-3051

# Summary
TeamPermission entity(s) and VimeoApiClient method(s).

## Description
Addition of entities and the calls which require them to VimeoApiClient

## How to Test
1. Set networking library version to "MPS-3051-team-permission-SNAPSHOT"
2. You can verify calls with this code snippet when logging in as the iap2@gmail.com test user:

```        
adaptRequest<TeamPermissionList> {
   vimeoApiClient.fetchTeamPermissions(
      uri = "/users/79937352/projects/5826885/team_permissions",
      fieldFilter = null,
      queryParams = null,
      cacheControl = null,
      callback = it
   )
}.subscribeBy {
   if(it is VimeoResponse.Success){
     Log.i("TEAMPERMISSIONS:" + it.data.toString())
   }
}.addTo(compositeDisposable)
```

### CODE REVIEW

File: models/src/main/java/com/vimeo/networking2/TeamPermission.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.vimeo.networking2
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+/**
+ * Represents the current permission relationship a [teamEntity] has with a resource. This class expresses any current
+ * way the [TeamEntity] is associated with the resource through [currentPermissions].  This class expresses any way the
+ * [TeamEntity] can have it's permission relationship to a resource edited via the set of potential permissions listed
```
Comment (anthonycr) [inline models/src/main/java/com/vimeo/networking2/TeamPermission.kt]: `it's` -> `its`. Can we add parameter documentation for the properties of this class?

File: models/src/main/java/com/vimeo/networking2/TeamPermissionCurrentPermissions.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+/**
+ * Represents the current permission for a [TeamPermission]. This only contains the uri for said permission as denoted
+ * by [permissionPolicyUri]
+ */
+@JsonClass(generateAdapter = true)
+data class TeamPermissionCurrentPermissions(
```
Comment (anthonycr) [inline models/src/main/java/com/vimeo/networking2/TeamPermissionCurrentPermissions.kt:12]: Needs a `param` doc for `permissionPolicyUri`.

**Review by anthonycr — APPROVED** [review]
Nice job, only thing I saw is there are a few missing param docs that should be added.

File: models/src/main/java/com/vimeo/networking2/TeamPermission.kt (hunk 1 above)
Comment (elhicks) [reply models/src/main/java/com/vimeo/networking2/TeamPermission.kt]: The parameters are explained inline by being mentioned in the description. I'll move them to params.

File: models/src/main/java/com/vimeo/networking2/TeamPermissionCurrentPermissions.kt (hunk 2 above)
Comment (elhicks) [reply models/src/main/java/com/vimeo/networking2/TeamPermissionCurrentPermissions.kt:12]: Once again, this is inlined in the description, but I'll move it out to a param.

**Pull request merged**